Utilities/DataSampling/src/Dispatcher.cxx

```diff
@@ -87,7 +88,7 @@ void Dispatcher::run(ProcessingContext& ctx)
     if (firstPart.header == nullptr) {
       continue;
     }
-    const auto* firstInputHeader = header::get<header::DataHeader*>(firstPart.header);
+    const auto* firstInputHeader = DataRefUtils::getHeader<header::DataHeader*>(firstPart);
```

Comment:
@knopers8 , please confirm that the changes are without effect to the functionality